### DISCUSSION TITLE
Phoenix Wallet Server Connection

### DISCUSSION THREAD

**Discussion by resignedScientist** [btcpayserver/btcpayserver#5908] [Plugins & integrations]
With the release of [phoenix wallet server (phoenixd)](https://github.com/ACINQ/phoenixd/), it would be nice to support it in BTCPayServer as one of the Lightning wallet options. 

They also have a nice [API](https://phoenix.acinq.co/server/api) with everything one would need.

**Comment by chrisguida** [comment]
Please add this!! Would be epic to be able to run PhoenixD and BTCPay on Start9! https://github.com/start9labs/phoenixd-startos

**Comment by Hodladi** [comment]
A feature request and something that would enable your application to alot more people that want to have their own LN node but dont want the hustle with channel management is to implement Phoenixd Server from ACINQ as LN backend to BTCPay Server.

Im guessing this would be a rather simple implementation to release for you guys.
The Phoenixd Server has a real simple API and got all the functions you need for BTCPay Server, they already implemented BOLT12 support.

https://phoenix.acinq.co/server/api

Payments

Create Bolt11 invoice
Get Bolt12 offer
Pay Bolt11 invoice
Pay Bolt12 offer
Pay on-chain address
List incoming payments
Get incoming payment
List outgoing payments
Get outgoing payment
Notifications API

Websocket
Webhook
Node management

Get node info
Get balance
List channels
Close channel
Decode invoice
Decode offer
Thanks in advance!

**Reply by maxmanbtc** [reply]
This would be great!

**Comment by pm47** [comment]
Is my understanding correct that what's needed is adding `phoenixd` as a backend in [Ride-The-Lightning](https://github.com/Ride-The-Lightning/RTL)?

**Reply by Hodladi** [reply]
No, Phoenixd Server is a Lightning Node it self. 
Its an alternative to LND, CLN) or Eclair

The difference is that ACINQ using their node and charge a fee on transactions for those who wanna use Phoenixd Server. But its still self custodial like running an LND node. Best part is that the person running the server doesnt need to hustle with channel balancing and even setting up channels.

So another LN backend for BTCPay Server just like the above mentioned.

Read more about it at their site
[https://phoenix.acinq.co/server](https://phoenix.acinq.co/server)

**Reply by pm47** [reply]
I know I'm the main dev of phoenixd. AFAICT btcpayserver relies on RTL for LN node integration, just want to confirm.

**Reply by Hodladi** [reply]
Oh, no, thats wrong.
btcpayserver connects via REST Proxy
![image](https://github.com/btcpayserver/btcpayserver/assets/91490683/966a6f69-64a0-451a-9729-9b5bc284edd1)

**Reply by Hodladi** [reply]
Or Eclair like this
![image](https://github.com/btcpayserver/btcpayserver/assets/91490683/988dbb26-02fd-435d-aac4-e4caf8c81fee)

**Reply by Hodladi** [reply]
Or C-Lightning like this
![image](https://github.com/btcpayserver/btcpayserver/assets/91490683/f58daac2-6ed0-4e9e-8c85-8c15d80fdc64)

**Reply by Hodladi** [reply]
I would guess that the eclair implementation would be similar to integrate phoenixd
Just a https or http with a password and then of course the correct API calls for it.

**Comment by JamesTsetsekas** [comment]
Using Phoenixd server as the LN back-end for BTCPay Server would be a great way to enable many people to run a node without worrying about channel management. I'm all for it.

**Comment by pavlenex** [comment]
There's some progress on this here https://github.com/btcpayserver/btcpayserver-docker/pull/987

**Reply by Sjors** [reply]
Nice! I don't use Docker, but it looks like they have a plugin coming soon(tm)...
